File: java/src/test/java/com/auth/APIKeyTest.java
package com.auth;

import com.auth.base.Authentication;
import com.auth.apikey.APIKey;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.springframework.http.HttpMethod;
import org.springframework.http.HttpStatus;
import org.springframework.http.RequestEntity;

import java.util.HashMap;
import java.util.Map;

import static org.junit.jupiter.api.Assertions.*;

class APIKeyTest extends TestUtils {
    private Authentication auth;

    @BeforeEach
    void setUp() {
        super.setUp();
        auth = new APIKey("valid-api-key");
    }

    @Test
    void testValidApiKeyInHeader() {
        Map<String, String> headers = new HashMap<>();
        headers.put("X-API-Key", "valid-api-key");
        RequestEntity<Void> request = createTestRequest(HttpMethod.GET, headers);
        runAuthTest(auth, request, HttpStatus.OK, false);
    }

    @Test
    void testValidApiKeyInQuery() {
        RequestEntity<Void> request = createTestRequest(HttpMethod.GET, null);
        request = new RequestEntity<>(request.getHeaders(), request.getMethod(), 
            request.getUrl().resolve("?api_key=valid-api-key"));
        runAuthTest(auth, request, HttpStatus.OK, false);
    }

    @Test
    void testInvalidApiKey() {
        Map<String, String> headers = new HashMap<>();
        headers.put("X-API-Key", "invalid-api-key");
        RequestEntity<Void> request = createTestRequest(HttpMethod.GET, headers);
        runAuthTest(auth, request, HttpStatus.UNAUTHORIZED, true);
    }

    @Test
    void testMissingApiKey() {
        RequestEntity<Void> request = createTestRequest(HttpMethod.GET, null);
        runAuthTest(auth, request, HttpStatus.UNAUTHORIZED, true);
    }

    @Test
    void testGetType() {
        assertAuthType(auth, "apikey");
    }
} 

File: php/tests/APIKeyTest.php
<?php

namespace Tests\Auth;

use Auth\APIKey;

class APIKeyTest extends TestUtils
{
    private $auth;

    protected function setUp(): void
    {
        parent::setUp();
        $this->auth = new APIKey('valid-api-key');
    }

    public function testValidApiKeyInHeader(): void
    {
        $request = $this->createTestRequest('GET', [
            'X-API-Key' => 'valid-api-key'
        ]);
        $this->runAuthTest($this->auth, $request, 200, false);
    }

    public function testValidApiKeyInQuery(): void
    {
        $request = $this->createTestRequest('GET');
        $request = $request->withUri($request->getUri()->withQuery('api_key=valid-api-key'));
        $this->runAuthTest($this->auth, $request, 200, false);
    }

    public function testInvalidApiKey(): void
    {
        $request = $this->createTestRequest('GET', [
            'X-API-Key' => 'invalid-api-key'
        ]);
        $this->runAuthTest($this->auth, $request, 401, true);
    }

    public function testMissingApiKey(): void
    {
        $request = $this->createTestRequest('GET');
        $this->runAuthTest($this->auth, $request, 401, true);
    }

    public function testGetType(): void
    {
        $this->assertAuthType($this->auth, 'apikey');
    }
} 